### PR TITLE
feat: add `fullWithSeconds` datetime format

### PR DESCRIPTION
**Pull request by andipaetzold**
Adds a new DateTime format `full-with-seconds`

This is very similar to the default `full` format but also displays seconds.
I am not happy with the name, so let me know if you have a better idea.

### CODE REVIEW

**Comment by fabe** [comment]
Maybe we could have use-case-based naming? e.g. `precise` or `exact`?

**Comment by andipaetzold** [comment]
> Maybe we could have use-case-based naming? e.g. `precise` or `exact`?

@fabe That might cause issues if we somewhere want to display milliseconds as well. But I agree that `precise` sounds better than `with-seconds`.

In general I love the [DateTimeFormat interface](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) where you enable parts with boolean flags. Having something like this in forma with the flags `date`, `time`, `seconds`, `weekday` would be a great interface and avoid naming issues. But I guess it's too late for discussing this now 😬

**Review by bgutsol — DISMISSED** [review]
`forma-v4` branch was already merged to master. Please change the base to `master`

**Review by bgutsol — APPROVED** [review]
LGTM

File: packages/components/datetime/src/types.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
 export type DateType = Date | string | number;
 
-export type DateFormat = 'full' | 'time' | 'weekday' | 'day';
+export type DateFormat =
+  | 'full-with-seconds'
```
Comment (gui-santos) [inline packages/components/datetime/src/types.ts:4]:
```suggestion
  | 'fullWithSeconds'
```

For values of props we prefer to use camelCase

**Review by gui-santos — CHANGES_REQUESTED** [review]
Looks good

I will only ask you to make the value camelCase, that's what we've been using for props
(e.g. spacingXl, colorBlue, endIcon, etc)

**Comment by andipaetzold** [comment]
@bgutsol @gui-santos Both happy with `fullWithSeconds` or should we go for `fullExact` as @fabe proposed?

**Comment by bgutsol** [comment]
> fullWithSeconds

Don't have a strong preference, but `fullWithSeconds` sounds more clear to me

**Comment by bgutsol** [comment]
> @bgutsol @gui-santos Both happy with `fullWithSeconds` or should we go for `fullExact` as @fabe proposed?

Don't have a strong preference, but fullWithSeconds sounds more clear to me

**Comment by gui-santos** [comment]
> @bgutsol @gui-santos Both happy with `fullWithSeconds` or should we go for `fullExact` as @fabe proposed?
@andipaetzold 
I have no preference tbh 🤷 
both sound good to me

**Comment by fabe** [comment]
I'd go for `fullWithSeconds` for now!

**Review by bgutsol — COMMENTED** [review]
@andipaetzold please, update from the latest master to get ci: check links fixed

**Review by suevalov — APPROVED** [review]

**Comment by andipaetzold** [comment]
I'd need an approval in chromatic in order to merge this PR

**Review by gui-santos — APPROVED** [review]

**Comment by gui-santos** [comment]
> I'd need an approval in chromatic in order to merge this PR

@andipaetzold done
Thanks a lot for this contribution 🙌

**Comment by andipaetzold** [comment]
Any idea why the released types are not up to date?

https://unpkg.com/@contentful/f36-datetime@4.1.0/dist/types.d.ts

`DateFormat` is missing the new `fullWithSeconds` option

**Comment by suevalov** [comment]
@andipaetzold is it only types or the code is also missing?

**Comment by andipaetzold** [comment]
@suevalov The code itself is there.

(Search for `fullWithSeconds` in https://unpkg.com/browse/@contentful/f36-datetime@4.1.0/dist/main.js)

**Comment by bgutsol** [comment]
> @suevalov The code itself is there.
> 
> (Search for `fullWithSeconds` in https://unpkg.com/browse/@contentful/f36-datetime@4.1.0/dist/main.js)

@andipaetzold  we redeployed it, and now it has proper types in v `4.1.1`:  https://unpkg.com/@contentful/f36-datetime@4.1.1/dist/types.d.ts

**Pull request merged**